### DISCUSSION TITLE
Single web adaptor with multimachine deployment

### DISCUSSION THREAD

**Discussion by sredar-fws** [Esri/arcgis-powershell-dsc#556] [Q&A] [answered]
It isn't clear to me how to configure the JSON file to use a single web adaptor with the multi-machine deployment.  In our case we have a portal machine with two server machines.

**Comment by cameronkroeker** [comment, marked as answer]
Hi @sredar-fws,

The AllNodes section is where you will want to define each machine and the role(s) of that machine.

**In the example below we have a distributed deployment where one role is assigned to each machine:**

- Machine 1: Portal for ArcGIS
- Machine 2: ArcGIS Server 1
- Machine 3: ArcGIS Server 2
- Machine 4: ArcGIS Web Adaptor (one registered with portal and one registered with server)
- Machine 5: ArcGIS Data Store registered as the relational/tilecache datastore with server.


```json
"AllNodes": [
        {
            "NodeName": "[Machine Name 1]",
            "Role": [
                "Portal"
            ]
        },
        {
            "NodeName": "[Machine Name 2]",
            "Role": [
                "Server"
            ]
        },
        {
            "NodeName": "[Machine Name 3]",
            "Role": [
                "Server"
            ]
        },
        {
            "NodeName": "[Machine Name 4]",
            "Role": [
                "WebAdaptor"
            ],
            "WebAdaptorConfig":[
                {
                    "Role": "Portal",
                    "Context": "arcgis"
                },
                {
                    "Role": "Server",
                    "Context": "serveradmin",
                    "AdminAccessEnabled": true,
                    "WebSiteId": 1
                }
            ]
        },
        {
            "NodeName": "[Machine Name 5]",
            "DataStoreTypes": [
                "Relational",
                "TileCache"
            ],
            "Role": [
                "DataStore"
            ]
        }
]
```

**Reply by sredar-fws** [reply]
Would adding a 2nd ArcGIS Server to the web adaptor config then look something like this?  And would the server order in the Node list be respected in terms the adaptor config?
```
         {
            "NodeName": "[machine name]",
            "Role": [
                "WebAdaptor"
            ],
            "WebAdaptorConfig":[
                {
                    "Role": "Portal",
		    "Context": "portal"
                },
                {
                    "Role": "Server",
		    "Context": "server",
		    "AdminAccessEnabled": true,
                    "WebSiteId": 1
                },
		{	
                    "Role": "Server",
		    "Context": "image",
		     "AdminAccessEnabled": true,
                     "WebSiteId": 2
		}
            ]
        }
```

**Reply by cameronkroeker** [reply]
This will register 2 web adaptors with the same ArcGIS Server site. One will be named "server" and the other will be named "image".

- https://[Web Adaptor FQDN]/server/admin
- https://[Web Adaptor FQDN]/image/admin

I believe both WebSiteId's should be set to 1, unless you are pre-configuring your IIS machine with multiple websites.
